crate.ts

```diff
@@ -83,9 +83,25 @@ export class Crate {
 
   async setVersion(version: string) {
     $.logStep(`Setting ${this.name} to ${version}...`);
-    for (const crate of this.repo.crates) {
-      await crate.setDependencyVersion(this.name, version);
+
+    if (this.repo.folderPath !== this.folderPath) {
+      const rootpath = $.path.join(this.repo.folderPath, "Cargo.toml");
+      const originalText = await Deno.readTextFile(rootpath);
+      const findRegex = new RegExp(
+        `^(\\b${this.name}\\b\\s.*)"([=\\^])?[0-9]+[^"]+"`,
+        "gm",
+      );
+
+      if (findRegex.test(originalText)) {
+        const newText = originalText.replace(findRegex, `$1"${version}"`);
+
+        if (originalText === newText) {
```

Comment:
It's probably sufficient to remove the `findRegex.test` condition above then do:

```ts
if (originalText !== newText) {
  await Deno.writeTextFile(rootpath, newText);
}
```

...since this if statement should never be true I think because of the test above.

Reply:
true